src/boot/efi/boot.c

```diff
@@ -2440,6 +2390,63 @@ static void save_selected_entry(const Config *config, const ConfigEntry *entry)
                 (void) efivar_set(LOADER_GUID, L"LoaderEntryLastBooted", NULL, EFI_VARIABLE_NON_VOLATILE);
 }
 
+static EFI_STATUS secure_boot_discover_keys(Config *config, EFI_FILE *root_dir) {
+        EFI_STATUS err;
+        _cleanup_(file_closep) EFI_FILE *keys_basedir = NULL;
+
+        if (secure_boot_mode() != SECURE_BOOT_SETUP)
+                return EFI_SUCCESS;
+
+        /* the lack of a 'keys' directory is not fatal and is silently ignored */
+        err = open_directory(root_dir, (CHAR16*) L"\\loader\\keys", &keys_basedir);
+        if (err == EFI_NOT_FOUND)
+                return EFI_SUCCESS;
+        if (EFI_ERROR(err))
+                return err;
+
+        for (;;) {
+                _cleanup_freepool_ CHAR16 *id = NULL, *title = NULL;
+                _cleanup_freepool_ EFI_FILE_INFO *dirent = NULL;
+                UINTN dirent_size = 0;
+                ConfigEntry *entry = NULL;
+
+                err = readdir_harder(keys_basedir, &dirent, &dirent_size);
+                if (!dirent)
+                        break;
+
+                if (EFI_ERROR(err))
+                        return err;
+
+                if (dirent->FileName[0] == '.')
+                        continue;
+
+                if (!FLAGS_SET(dirent->Attribute, EFI_FILE_DIRECTORY))
+                        continue;
+
+                id = xpool_print(L"secure-boot-keys-%s", dirent->FileName);
+                title = xpool_print(L"Enroll Secure Boot keys: %s", dirent->FileName);
+
+                entry = config_entry_add_call(
+                        config,
+                        LOADER_SECURE_BOOT_KEYS,
+                        id,
+                        title,
+                        secure_boot_enroll);
+                if (!entry)
+                        continue;
+                entry->path = xpool_print(L"\\loader\\keys\\%s", dirent->FileName);
+
+                /* if auto enrollment is activated and set to force, we try to load keys straight from the /loader/keys/auto location. */
```

Comment:
This comment is not needed. It just describes the code that is already pretty obvious to the reader, no?